[fltk] Update 1.4.0

hash2000 opened the pull request:
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version.~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

hash2000 committed:
[fltk] Update 1.4.0

hash2000 committed:
[fltk] Update 1.4.0

fix fluid-cmd and options-cmd installations

hash2000 committed:
Update version database

hash2000 committed:
Update ports/fltk/portfile.cmake

Co-authored-by: Cheney Wang <38240633+Cheney-W@users.noreply.github.com>

hash2000 commented:
@microsoft-github-policy-service agree

hash2000 committed:
Update version database

hash2000 commented:
I have such a question. I use this port in my pet project. I really want to upgrade to version 4+, because there are improvements that I need there. but I have no way to check why it is not being built in osx, android and linux environments. how do I ask for help from the community to complete my pool request

dg0yt commented:
You can get the build logs from CI: https://github.com/microsoft/vcpkg/discussions/31357

FLTK 1.4 is substantially different from the previous versions. It probably needs patches to be recreated from scratch.

hash2000 commented:
Yes, indeed, there are significant differences. but it doesn't seem very complicated. I see that I have stumbled into some kind of problem that is also not difficult to solve.  but this port is too good to ignore. I'll finish it!

hash2000 committed:
remove file ports/fltk/fix-system-link.patch

hash2000 committed:
add features X11, python3

hash2000 committed:
format fltk/vcpkg.json

hash2000 committed:
Update version database

hash2000 committed:
Merge branch 'master' into feature/fltk-1.4.0

hash2000 committed:
add dependence libxft

hash2000 committed:
add fix-export.patch to resolve error "bin/fltk-options-cmd does not exist"

hash2000 committed:
add fix-export.patch to resolve error "bin/fltk-options-cmd does not exist"

hash2000 committed:
fix copied tools list

Cheney-W commented:
@hash2000 Could you please solve the conflicts in `scripts/azure-pipelines/linux/provision-image.sh`? Thank you!

dg0yt commented on scripts/azure-pipelines/linux/provision-image.sh:
```diff
@@ -47,7 +47,7 @@ APT_PACKAGES="git curl zip unzip tar"
 ## common build dependencies
 APT_PACKAGES="$APT_PACKAGES at libxt-dev gperf libxaw7-dev cifs-utils \
-  build-essential g++ gfortran libx11-dev libxkbcommon-x11-dev libxi-dev \
+  build-essential gcc g++ gfortran libx11-dev libxkbcommon-x11-dev libxi-dev \
```
This file shall not be changed by this PR.

hash2000 replied:
No, it was an experiment.
I needed to create a separate branch for experiments so that they wouldn't end up in the pool request. 
My mistake

hash2000 committed:
remove fix-export

hash2000 committed:
remove tool installation fltk-options-cmd

hash2000 committed:
add libxft devel

hash2000 committed:
add libxft devel

hash2000 committed:
add libxft devel

hash2000 committed:
add libxft devel in CMake warning

hash2000 committed:
............................................________

....................................,.-'"...................``~.,
.............................,.-"..................................."-.,
.........................,/...............................................":,
.....................,?......................................................,
.................../...........................................................,}
................./......................................................,:`^`..}
.............../...................................................,:"........./
..............?.....__.........................................:`.........../
............./__.(....."~-,_..............................,:`........../
.........../(_...."~,_........"~,_....................,:`........_/
..........{.._$;_......"=,_......."-,_.......,.-~-,},.~";/....}
...........((.....*~_......."=-._......";,,./`..../"............../
...,,,___.`~,......"~.,....................`.....}............../
............(....`=-,,.......`........................(......;_,,-"
............/.`~,......`-...................................../
.............`~.*-,.....................................|,./.....,__
,,_..........}.>-._...................................|..............`=~-,
.....`=~-,__......`,.................................
...................`=~-,,.,...............................
................................`:,,...........................`..............__
.....................................`=-,...................,%`>--==``
........................................_..........._,-%.......`
...................................,

hash2000 committed:
remove experiments

hash2000 committed:
add cairo

hash2000 committed:
add pango

hash2000 committed:
fix dependencies

hash2000 committed:
cairo, pango is not default features

hash2000 committed:
next test

dg0yt commented on ports/fltk/portfile.cmake:
```diff
@@ -5,14 +5,13 @@ vcpkg_from_github(
+    SHA512 7630d0b02ff09c277d5641e10631514e5e3d8087e81f5254f38a8adb05c39ed0092ae81697085ed0dd859f0b826f94626d698090153c5e9a655f5e36263b2915
     PATCHES
-        dependencies.patch
+        fix-options.patch
```
Could you please rename `fix-options.patch` back to `dependencies.patch`?

hash2000 committed:
next test

hash2000 committed:
Merge branch 'master' into feature/fltk-1.4.0-experiments

hash2000 committed:
cairo, pango

hash2000 committed:
review

hash2000 committed:
next test

hash2000 committed:
next test

hash2000 committed:
add required libraries

hash2000 committed:
Merge branch 'master' into feature/fltk-1.4.0

hash2000 committed:
add required libraries for android and linux

hash2000 committed:
Update version database

hash2000 committed:
fix CMake error "Unexpected READ_ACCESS on variable WIN32 in script mode"

hash2000 committed:
Update version database

hash2000 committed:
fix message for requires packages

hash2000 committed:
Update version database

dg0yt reviewed (commented):
We don't need to discuss about mathgl before the port builds well enough. ATM mathgl fails because the CMake config is in invalid state.

hash2000 committed:
Update ports/fltk/portfile.cmake

Co-authored-by: Kai Pastor <dg0yt@darc.de>

hash2000 committed:
review

hash2000 committed:
Update version database

dg0yt commented on scripts/azure-pipelines/linux/provision-image.sh:
```diff
@@ -149,6 +149,11 @@ APT_PACKAGES="$APT_PACKAGES cuda-compiler-12-8 cuda-libraries-dev-12-8 cuda-driv
   libnccl2 libnccl-dev"
 
+## required by fltk
+APT_PACKAGES="$APT_PACKAGES libasound2-dev libglu1-mesa-dev libwayland-dev \
+    wayland-protocols libdbus-1-dev libxkbcommon-dev libegl-dev libopengl-dev \
+    libpango1.0-dev libgtk-3-dev"
```
You must add dependencies in vcpkg.json.

hash2000 replied:
a bit of reasoning. The dependencies of dev packages are made from vcpkg.json, probably so that it can be debugged. But. There is a libgtk-3-dev package in fltk for linux. if you add it to vcpkg.the json build of fltk on my laptop increases to an hour and a half instead of 15 minutes. and I know for sure that I don't need gtk debugging.

dg0yt commented on ports/fltk/portfile.cmake:
```diff
@@ -22,12 +21,16 @@ file(REMOVE_RECURSE
 
+if(VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX OR VCPKG_TARGET_IS_ANDROID)    
+    message(INFO "Building ${PORT} requires packages. Please run the following command.\nsudo apt-get update -y && sudo apt-get install -y libasound2-dev libglu1-mesa-dev libwayland-dev wayland-protocols libdbus-1-dev libxkbcommon-dev libegl-dev libopengl-dev libpango1.0-dev libgtk-3-dev")
+endif()
```
Message are useful and recommended. When they give the right advise. But there is no `apt` on macOS, and when you build for Android, the use of `apt` on the Linux host computer doesn't solve any problem for your target.

dg0yt commented:
As long as there are changes in `scripts/azure-pipelines` and not the required dependencies in `vcpkg.json`, there is no point in testing/reviewing.
